### PR TITLE
pybamm.plot_voltage_components doc update

### PR DESCRIPTION
**Pull request by Hongmeiqi**
add link of matplotlib docs for ax.fill_between function

# Description

Choose the task "pybamm.plot_voltage_components" in issue#3392 to do the hackathon practise for PyBaMM Developer Training Workshop.

### CODE REVIEW

**Review by Saransh-cpp — APPROVED** [review]
Thanks, @Hongmeiqi! Makes the docstring more informative.

**Comment by agriyakhetarpal** [comment]
Also, we can use Matplotlib's intersphinx inventory to link to the relevant attribute. @Hongmeiqi, could you add `"matplotlib": ("https://matplotlib.org/stable/", None)` to `intersphinx_mapping` in `docs/conf.py`? It would add the link for you and will be cleaner.

**Comment by Hongmeiqi** [comment]
> Also, we can use Matplotlib's intersphinx inventory to link to the relevant attribute. @Hongmeiqi, could you add `"matplotlib": ("https://matplotlib.org/stable/", None)` to `intersphinx_mapping` in `docs/conf.py`? It would add the link for you and will be cleaner.

Thanks Agriya! Seems it has been added to `intersphinx_mapping` in `docs/conf.py`. Good to know this way to add links.

**Comment by agriyakhetarpal** [comment]
Thanks! I think it should work if you change it to
````
:func:`matplotlib.axes.Axes.fill_between`
````

you may verify this by building the documentation locally (`nox -s docs`) or on the PR preview, here: https://pybamm--3993.org.readthedocs.build/en/3993/

**Comment by Hongmeiqi** [comment]
> Thanks! I think it should work if you change it to
> 
> ```
> :func:`matplotlib.axes.Axes.fill_between`
> ```
> 
> you may verify this by building the documentation locally (`nox -s docs`) or on the PR preview, here: https://pybamm--3993.org.readthedocs.build/en/3993/

Thank you Agriya. Just changed as you told. Can you help review again?

**Review by agriyakhetarpal — COMMENTED** [review]
I think this looks better now, we just need to figure out why it isn't linking to the Matplotlib docs and we should be good to go after that

**Comment by agriyakhetarpal** [comment]
`pipx run sphobjinv suggest -t 90 -u https://matplotlib.org/stable/ "matplotlib.axes.Axes.fill_between"`

gives me this output:

<details>
<summary>Tap to reveal</summary>

```
Attempting https://matplotlib.org/stable/ ...
  ... no recognized inventory.
Attempting "https://matplotlib.org/stable/objects.inv" ...
  ... inventory found.
 
------------------------------------------------------------------------

The intersphinx_mapping for this docset is LIKELY:

  (https://matplotlib.org/stable/, None)

------------------------------------------------------------------------

Project: Matplotlib
Version: 3.8.4

10433 objects in inventory.

------------------------------------------------------------------------

4 results found at/above current threshold of 90.

:py:method:`matplotlib.axes.Axes.fill_between`
:py:method:`matplotlib.axes.Axes.fill_betweenx`
:std:doc:`api/_as_gen/matplotlib.axes.Axes.fill_between`
:std:doc:`api/_as_gen/matplotlib.axes.Axes.fill_betweenx`
```

</details>

I guess `:py:method:`matplotlib.axes.Axes.fill_between` will work here.

File: pybamm/plotting/plot_voltage_components.py [hunk 1]
```diff
@@ -34,8 +34,7 @@ def plot_voltage_components(
         Whether to show the plots. Default is True. Set to False if you want to
         only display the plot after plt.show() has been called.
     kwargs_fill
-        Keyword arguments, passed to ax.fill_between
-
+        Keyword arguments : :func:`matplotlib.axes.Axes.fill_between`
```
Comment (agriyakhetarpal) [inline pybamm/plotting/plot_voltage_components.py:37]:
```suggestion
        Keyword arguments: :py:method:`matplotlib.axes.Axes.fill_between`
```

**Comment by Hongmeiqi** [comment]
> `pipx run sphobjinv suggest -t 90 -u https://matplotlib.org/stable/ "matplotlib.axes.Axes.fill_between"`
> 
> gives me this output:
> 
> Tap to reveal
> I guess
> 
> ```
> :py:method:`matplotlib.axes.Axes.fill_between`
> ```
> 
> will work here. Edit: it raises an error, so we can use the generic `:obj:` – verified the change locally and it's working.

Thanks Agriya! I changed it into obj and managed to use nox to verify it now!

**Review by agriyakhetarpal — APPROVED** [review]

**Comment by agriyakhetarpal** [comment]
@all-contributors please add @Hongmeiqi for docs

**Review by arjxn-py — APPROVED** [review]
Thanks @Hongmeiqi

**Pull request merged**